### PR TITLE
Further major refactoring

### PR DESCRIPTION
**Pull request by bcwaldon**
This PR aims to simplify the SDK by removing a number of elements that are no longer necessary.

1. The idea of a "build_env" is gone, since we don't expect users to use that containerized environment anymore. The image is retained (now called the "tools" image), but the build_env* targets are gone. A user simply inherits from the public docker images.
2. The sample applications are now coalesced into two very clear examples in `examples/app-*`. I am not sure the CPP approach is correct (will discuss offline) and I need to come back after this with better support for modes (as documented in examples/entrypoint.sh).
3. The app packaging automation has been removed, as a customer working with Docker should understand how to use base images and install their code appropriately. There are many examples of this between the examples here and the demos repo.
4. The pc-sim is removed, as we are going to focus on ACP and lib-driven testing. We will re-evaluate the need for a local "simulator" down the road

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,118 +5,29 @@ Specifically, this enables the development and testing of payload applications:
...
-### Launch Build Environment
+The development processes available here utilize Docker containers.
+It is expected that developers start from one of the supported base images:
 
-Use `make build_env` to instantiate a build environment container and open a new shell within it.
+* Python applications: `quay.io/antaris-inc/satos-payload-app-python:stable`
+* C++ application: quay.io/antaris-inc/satos-payload-app-cpp:stable
 
-NOTE: This step also builds the requisite Docker image which requires internet access to install some system packages.
+A developer does not even need to close this repository in order to build an application, however there are examples
```
Comment (rahulbhivare) [inline README.md:28]: Spelling mistake = "even need to **close** this repository" .. "even need to **clone** this repository"

File: examples/app-python/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+In the Antaris Cloud Platform, create a TrueTwin Satellite with a remote payload and download the associated config.
+Place that downloaded zip file in this directory.
+
+Build the app using the following command:
+
+```
+docker build --platform=linux/amd64 -t satos-payload-example-app-python .
+```
+
+Next, we can run the application in a container. The command below assumes that `CONFIG` is set to the name of the downloaded file in your current working directory:
```
Comment (rahulbhivare) [inline examples/app-python/README.md:17]: Meaning of "Downloaded file", we may need to explain.
Downloaded file = path of zip file downloaded for true-twin.

File: examples/app-python/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+```
+docker run --platform=linux/amd64 -e CONFIG=$CONFIG -v $(pwd):/workspace -it satos-payload-example-app-python
```
Comment (rahulbhivare) [inline examples/app-python/README.md:20]: I gave CONFIG as absolute oath of .zip file. But I got following error :

**$ export CONFIG=~/Downloads/_Packages_upload_RahulPayload-1684750606-aaa28e2d01.zip 
rahul@Rahul:~/projects/payload_git/test/SatOS-Payload-SDK/examples/app-python$ docker run --platform=linux/amd64 -e CONFIG=$CONFIG -v $(pwd):/workspace -it satos-payload-example-app-python
unzip:  cannot find or open /workspace//home/rahul/Downloads/_Packages_upload_RahulPayload-1684750606-aaa28e2d01.zip, /workspace//home/rahul/Downloads/_Packages_upload_RahulPayload-1684750606-aaa28e2d01.zip.zip or /workspace//home/rahul/Downloads/_Packages_upload_RahulPayload-1684750606-aaa28e2d01.zip.ZIP.**

In root-entrypoint.sh, CONFIG_ABS is calculated appending $pwd. 
So, we may need to specify explicitly about 'give relative path w.r.t. current dir' .

Comment (rahulbhivare) [inline examples/app-python/README.md:20]: After copying zip file in current dir and providing proper paths, I got following error =

**$ export CONFIG=_Packages_upload_RahulPayload-1684750606-aaa28e2d01

$ docker run --platform=linux/amd64 -e CONFIG=$CONFIG -v $(pwd):/home/rahul/projects/payload_git/test/SatOS-Payload-SDK -it satos-payload-example-app-python

unzip:  cannot find or open /workspace/_Packages_upload_RahulPayload-1684750606-aaa28e2d01, /workspace/_Packages_upload_RahulPayload-1684750606-aaa28e2d01.zip or /workspace/_Packages_upload_RahulPayload-1684750606-aaa28e2d01.ZIP.**

**Review by deepaktawri — APPROVED** [review]

File: lib/python/satos_payload_sdk/antaris_api_gpio.py [hunk 4]
```diff
@@ -76,3 +76,26 @@ def api_pa_pc_write_gpio(pin, value):
     time.sleep(g_SLEEP_TIME_IN_SEC)
     op = (Device.port >> pin) & g_MASK_BIT_0
```
Comment (rahulbhivare) [inline lib/python/satos_payload_sdk/antaris_api_gpio.py:77]: This file provides api to make gpio pins high/low.
In Janus-1, one of payload wanted "high-low-high" pulse. So, this file provides api for that.

File: lib/python/satos_payload_sdk/antaris_api_gpio.py [hunk 5]
```diff
@@ -76,3 +76,26 @@ def api_pa_pc_write_gpio(pin, value):
     time.sleep(g_SLEEP_TIME_IN_SEC)
```
Comment (rahulbhivare) [inline lib/python/satos_payload_sdk/antaris_api_gpio.py:76]: This is example code of 'how to use gpio pins'

**Review by deepaktawri — COMMENTED** [review]
I imagine the intention is to have a generic folder name "tools" (instead of build-tools). Would it make sense to move all these files in a sub-folder "tools/build/" to categorise ? In future, more stuff can be added in tools directory which may not be build specific.

**Review by deepaktawri — COMMENTED** [review]
1. Now that build-env target is gone, its not clear how to make changes in build environment container image and overwrite/upload existing image.

2. sdk/app_pkg targets were meant to create debian package to be installed in payload VMs. How that will be done now ?

**Review by deepaktawri — COMMENTED** [review]
examples is better than samples. However, examples/app/cpp, examples/app/python kind of subfolder structure would have been more standard. I am not strong about it though.

**Review by deepaktawri — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (bcwaldon) [reply README.md:28]: Yes, good catch!

File: examples/app-python/README.md (hunk 2 above)
Comment (bcwaldon) [reply examples/app-python/README.md:17]: Cleaned this up.

File: examples/app-python/README.md (hunk 3 above)
Comment (bcwaldon) [reply examples/app-python/README.md:20]: Yes, that is right. I have fixed this in the docs.

File: lib/python/satos_payload_sdk/antaris_api_gpio.py (hunk 4 above)
Comment (bcwaldon) [reply lib/python/satos_payload_sdk/antaris_api_gpio.py:77]: Understood. I am just placing the example usage in the file for now until we have a chance to place it more appropriately.

**Comment by bcwaldon** [comment]
Thanks for input @deepaktawri  and @rahulbhivare . Let's continue the discussion about packaing for VM targets vs containers and introduce that support here when necessary. On the folder nesting structure, I am a fan of keeping things simple until the nesting is necessary. The images/ dir, for example,map directly to the name of the images published to quay.io, for example. 

I am merging this now since the changes are disruptive, and we can iterate on master from here.

**Pull request merged**